src/MongoDB.Driver/Linq/MongoQueryable.cs

```diff
@@ -3416,72 +3189,30 @@ public static Task<long> SumAsync<TSource>(this IMongoQueryable<TSource> source,
                     GetMethodInfo(Take, source, count),
-                    Expression.Convert(source.Expression, typeof(IMongoQueryable<TSource>)),
+                    source.Expression,
                     Expression.Constant(count)));
         }
 
         /// <summary>
-        /// Performs a subsequent ordering of the elements in a sequence in ascending
-        /// order according to a key.
+        /// 
         /// </summary>
-        /// <typeparam name="TSource">The type of the elements of <paramref name="source" />.</typeparam>
-        /// <typeparam name="TKey">The type of the key returned by the function that is represented by keySelector.</typeparam>
-        /// <param name="source">A sequence of values to order.</param>
-        /// <param name="keySelector">A function to extract a key from an element.</param>
-        /// <returns>
-        /// An <see cref="IOrderedMongoQueryable{TSource}"/> whose elements are sorted according to a key.
-        /// </returns>
-        public static IOrderedMongoQueryable<TSource> ThenBy<TSource, TKey>(this IOrderedMongoQueryable<TSource> source, Expression<Func<TSource, TKey>> keySelector)
+        /// <typeparam name="TDocument"></typeparam>
+        /// <param name="source"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public static Task<List<TDocument>> ToListAsync<TDocument>(this IQueryable<TDocument> source, CancellationToken cancellationToken = default(CancellationToken))
         {
-            Ensure.IsNotNull(source, nameof(source));
-            Ensure.IsNotNull(keySelector, nameof(keySelector));
-
-            return (IOrderedMongoQueryable<TSource>)Queryable.ThenBy(source, keySelector);
-        }
-
-        /// <summary>
-        /// Performs a subsequent ordering of the elements in a sequence in descending
-        /// order according to a key.
-        /// </summary>
-        /// <typeparam name="TSource">The type of the elements of <paramref name="source" />.</typeparam>
-        /// <typeparam name="TKey">The type of the key returned by the function that is represented by keySelector.</typeparam>
-        /// <param name="source">A sequence of values to order.</param>
-        /// <param name="keySelector">A function to extract a key from an element.</param>
-        /// <returns>
-        /// An <see cref="IOrderedMongoQueryable{TSource}"/> whose elements are sorted in descending order according to a key.
-        /// </returns>
-        public static IOrderedMongoQueryable<TSource> ThenByDescending<TSource, TKey>(this IOrderedMongoQueryable<TSource> source, Expression<Func<TSource, TKey>> keySelector)
-        {
-            Ensure.IsNotNull(source, nameof(source));
-            Ensure.IsNotNull(keySelector, nameof(keySelector));
-
-            return (IOrderedMongoQueryable<TSource>)Queryable.ThenByDescending(source, keySelector);
+            var cursorSource = GetCursorSource(source, nameof(ToListAsync));
```

Comment:
`source` is of type `IQueryable<T>` which does *not* implement `IAsyncCursorSource<T>`, but... our class `MongoQuery` that is our implementation of `IQueryable<T>` ALSO implements `IAsyncCursorSource<T>`.